book/src/writing-apps/overview.md

```diff
@@ -26,7 +26,12 @@ For more information on both commands, see the [build](./build.md) docs.
 
 ### Inputs
 
-The `--input` field needs to either be a hex string or a file path to a file that will be read as bytes. Note that if your hex string represents a single number, it should be written in little-endian format (as this is what the VM expects). To see how more complex inputs can be converted into a VM-readable format, see the **Using StdIn** section of the [SDK](../advanced-usage/sdk.md) doc.
+The `--input` field needs to either be a hex string or a file path to a json file that contains the key `input` and an array of hex strings. Note that if your hex string represents a single number, it should be written in little-endian format (as this is what the VM expects).
```

Comment:
emphasize a little more that if its a hex string, that is a single hex string, while with json file, it is an _array_ of hex string.

also I am still a bit surprised how `--input` can distinguish hex string from filepath. if it's easy, I feel like distinguishing with `--json` vs `--input` might be better, but only if it's easy